### ISSUE TITLE
[6.3] non-existent resolver Symfony\Bridge\Doctrine\ArgumentResolver\EntityValueResolver

### ISSUE THREAD

**Issue by PhilETaylor** [symfony/symfony#50081]
### Symfony version(s) affected

6.3.x-dev

### Description

I had a fully working 6.2 app and I upgraded to 6.3.x-dev as a test.

My controller has some new (and working in 6.2) `MapEntity` attributes like

```php
 #[MapEntity(mapping: [
            'location_id' => 'id',
        ])] Locations $location,
```

Since `[HttpKernel] Introduce pinnable value resolvers with #[ValueResolver] and #[AsPinnedValueResolver]` #48992 was merged I now get this error 

<img width="1191" alt="ScreenShot-2023-04-20-18 06 50" src="https://user-images.githubusercontent.com/400092/233438263-dd87498d-4cd5-4efc-844f-09bdd8f906b4.png">

Sorry way beyond me... 

Ping @MatTheCat 

### How to reproduce

Here is a reproducer - https://github.com/PhilETaylor/mapentity_reproducer

To create this reproducer I did 

- composer create-project symfony/skeleton bug_app
- change composer.json to allow 6.3.x and dev stability
- composer require orm 
- composer require symfony/maker-bundle --dev
- use maker bundle to create a controller - edit the controller to add the MapEntity
- use maker bundle to create an entity
- symfony serve
- go to http://127.0.0.1:8000/bug - you can see the error there

### Possible Solution

Pay @MatTheCat some sponsorship... incoming.... Edit: No sponsor button :-( 

### Additional Context

I can "fix" this for my own purposes with this in services.php so maybe its a tagging issue?

```php
$services->set(\Symfony\Bridge\Doctrine\ArgumentResolver\EntityValueResolver::class)
        ->tag('doctrine.orm.entity_value_resolver');
```

**Comment by MatTheCat** [comment]
Maybe your DoctrineBundle does not include https://github.com/doctrine/DoctrineBundle/pull/1624? It was added in the 2.9.0 version.

**Comment by PhilETaylor** [comment]
I can confirm that the file at: `vendor/doctrine/doctrine-bundle/Resources/config/orm.xml` in the reproducer

contains

```xml
        <service id="doctrine.orm.entity_value_resolver" class="Symfony\Bridge\Doctrine\ArgumentResolver\EntityValueResolver">
            <argument type="service" id="doctrine" />
            <argument type="service" id="doctrine.orm.entity_value_resolver.expression_language" on-invalid="ignore" />
            <tag name="Symfony\Bridge\Doctrine\ArgumentResolver\EntityValueResolver" priority="110">controller.argument_value_resolver</tag>
        </service>
```

The composer.json having contained `"doctrine/doctrine-bundle": "2.10.x-dev",`

**Comment by MatTheCat** [comment]
Okay it seems it has to do with #36586 as the `name` attribute of XML defined tags is not returned by `Definition::getTag`. Does it ring a bell @nicolas-grekas?

**Comment by MatTheCat** [comment]
~It seems the right syntax for adding attributes to tags using XML is not the one mentioned in #36586~

The syntax mentioned in #36586 does no longer work, but the following does:

```xml
<tag name="controller.argument_value_resolver" priority="110">
    <attribute name="name">Symfony\Bridge\Doctrine\ArgumentResolver\EntityValueResolver</attribute>
</tag>
```

Also [the documentation](https://symfony.com/doc/current/service_container/tags.html#adding-additional-attributes-on-tags) is wrong because an `attribute` tag without a `name` attribute would trigger an exception…

**Comment by stof** [comment]
The documentation is not wrong actually. There is only a special case when wanting to add an attribute named `name` in XML to avoid conflicts with the tag name.

**Comment by stof** [comment]
however, the documentation should probably cover that special case, which is not the case right now.

**Comment by MatTheCat** [comment]
Given https://github.com/symfony/symfony/blob/24b1b022dceb7369a68d44b5487567fd1c5dc0d3/src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php#L658-L663 wouldn’t the documentation code crash?

```xml
<tag name="app.mail_transport">
    <attribute>sendmail</attribute>
    <attribute>anotherAlias</attribute>
</tag>
```

I thought #36586 allowed to have a `name` attribute as long as the tag name was the `tag` tag text content :thinking:

**Comment by stof** [comment]
Ok, apparently, the documentation was updated in the wrong way after https://github.com/symfony/symfony/pull/36586 (contrary to the comment on https://github.com/symfony/symfony-docs/issues/13613 saying that the doc did not needed any update)

**Comment by MatTheCat** [comment]
Seems #36586 is broken for XML since 6.2, probably because of #47364

**Comment by MatTheCat** [comment]
I opened #50088 to illustrate the cause of this bug.

The actual state of the documentation is another issue (`<attribute>` tags are missing a `name` XML attribute).

**Comment by PhilETaylor** [comment]
Thanks for your time investigating this. Really appreciate it personally, but seems like it could have been a bigger issue if it had not been caught now.

**Comment by PhilETaylor** [comment]
Closing as fixed by proposed PR #50088

**Comment by MatTheCat** [comment]
I don’t think the issue would have been bigger (one patch version away?), but it’s indeed better if features added in 6.3 work when it is released :grin: 

So thanks for testing!
